src/vm/domainfile.cpp

```diff
@@ -593,6 +596,45 @@ BOOL DomainFile::DoIncrementalLoad(FileLoadLevel level)
     return TRUE;
 }
 
+void DomainAssembly::VerifyReadyToRunImageDependencies()
+{
+    //
+    // Verify that the IL image is consistent with the R2R images loaded into appdomain
+    //
+
+    AssemblySpec spec;
+    spec.InitializeSpec(GetFile());
+
+    GUID mvid;
+    GetFile()->GetMVID(&mvid);
+
+    GetAppDomain()->CheckForMismatchedNativeImages(&spec, &mvid);
+    
+    if (GetFile()->IsILImageReadyToRun())
+    {
+        PTR_PEImage pImage = GetFile()->GetILimage();
+
+        PTR_PEImageLayout pLayout = pImage->GetLayout(PEImageLayout::LAYOUT_ANY, 0);
+        PTR_IMAGE_DATA_DIRECTORY pDependencyDirectory = pLayout->GetReadyToRunSection(READYTORUN_SECTION_NATIVE_DEPENDENCIES);
+        if (pDependencyDirectory != NULL)
+        {
+            READYTORUN_DEPENDENCY* pDependencies = (READYTORUN_DEPENDENCY*)pLayout->GetDirectoryData(pDependencyDirectory);
+            COUNT_T cDependencies = pDependencyDirectory->Size / sizeof(READYTORUN_DEPENDENCY);
+            for (COUNT_T iDependency = 0; iDependency < cDependencies; iDependency++)
+            {
+                READYTORUN_DEPENDENCY* pDependency = &(pDependencies[iDependency]);
+
+                AssemblySpec name;
+                name.InitializeSpec(pDependency->AssemblyToken,
+                    pImage->GetNativeMDImport(),
+                    GetDomainAssembly());
+
+                GetAppDomain()->CheckForMismatchedNativeImages(&name, &pDependency->Mvid);
```

Comment:
The CheckForMismatchedNativeImages feature doesn't appear to me to be correct in the presence of AssemblyLoadContext's with non-default behavior which load ReadyToRun images.

There are 2 issues
1. The list is AppDomain level, and that's not appropriate for checks that do not have AppDomain locality.
2. There is assembly name normalization that occurs as part of CheckForMismatchedNative images. This normalization is appropriate for the non-AssemblyLoadContext world, but in the presence of assembly load contexts, normalization isn't correct.